common/network-common/src/main/java/org/apache/spark/network/util/TransportFrameDecoder.java

```diff
@@ -139,14 +139,18 @@ private ByteBuf decodeNext() throws Exception {
       return nextBufferForFrame(remaining);
     }
 
-    // Otherwise, create a composite buffer.
-    CompositeByteBuf frame = buffers.getFirst().alloc().compositeBuffer();
+    // Otherwise, create a new buffer to hold the data of all frame splits.
+    ByteBuf frame = buffers.getFirst().alloc().buffer(remaining, remaining);
```

Comment:
here allocate a new `ByteBuf`, it will result in one time copy for each netty frame, exactly one time copy operation is much more acceptable compared with too many copies of `compositeBuffer`.